pkg/util/errors/errors.go

```diff
@@ -29,10 +29,16 @@ type Aggregate interface {
 // is itself an implementation of the error interface.  If the slice is empty,
 // this returns nil.
 func NewAggregate(errlist []error) Aggregate {
-	if len(errlist) == 0 {
+	errs := []error{}
+	for _, err := range errlist {
+		if err != nil {
```

Comment:
This is interesting - a list of nil errors now gets treated as a non-error.  I am not sure that is right.  Why not substitute `fmt.Errorf("(nil)")`.  This seems more consistent with Go's assertion that a typed nil error is an error, and not nil.